src/towncrier/test/test_builder.py

```diff
@@ -132,3 +134,73 @@ def test_orphan_all_digits(self):
             parse_newfragment_basename("+123.feature", ["feature"]),
             ("+123", "feature", 0),
         )
+
+
+class TestIssueOrdering(TestCase):
+    """
+    Tests to ensure that issues are ordered correctly in the output.
+
+    This tests both ordering of issues within a fragment and ordering of
+    fragments within a section.
+    """
+
+    template = dedent(
+        """
+    {% for section_name, category in sections.items() %}
+    {% if section_name %}# {{ section_name }}{% endif %}
+    {%- for category_name, issues in category.items() %}
+    ## {{ category_name }}
+    {% for issue, numbers in issues.items() %}
+    - {{ issue }}{% if numbers %} ({{ numbers|join(', ') }}){% endif %}
+
+    {% endfor %}
+    {% endfor -%}
+    {% endfor -%}
+    """
+    )
+
+    def render(self, fragments):
+        return render_fragments(
+            template=self.template,
+            issue_format=None,
+            fragments=fragments,
+            definitions={},
+            underlines=[],
+            wrap=False,
+            versiondata={},
+        )
+
+    def test_ordering(self):
+        """
+        Issues are ordered first by the non-text component, then by their number.
```

Comment:
I guess here "issues" means "news fragments" 


```suggestion
        News fragments are ordered first by the non-text component, then by their number.
```